### PR TITLE
DR-389 Fixed up study delete validation; moved file tests

### PR DESCRIPTION
**Pull request by ddietterich**
I hooked up the metadata-based study-dataset dependency check.
I changed the filesystem-level dependency check to be a corruption detection: if there are stray file dependencies we will detect them and barf.

I also brought all of the filesystem tests together in the `filesystem` package, moving some from the `controller` package.

### CODE REVIEW

File: src/main/java/bio/terra/flight/study/delete/DeleteStudyValidateStep.java [hunk 1]
```diff
@@ -1,28 +1,40 @@
...
+        List<DatasetSummary> datasets = datasetDao.retrieveDatasetsForStudy(studyId);
+        if (datasets.size() != 0) {
+            throw new ValidationException("Can not delete a study being used by datasets");
+        }
+        // Sanity check - validate that there are no stray file references. There should be none left
+        // if there are no datasets returned from retrieveDatasetsForStudy.
         if (dependencyDao.studyHasDatasetReference(studyId.toString())) {
-            throw new ValidationException("Can not delete study being used by datasets");
+            throw new FileSystemCorruptException("File system has study dependencies; metadata does not");
         }
```
Comment (RoriCremer) [inline src/main/java/bio/terra/flight/study/delete/DeleteStudyValidateStep.java:38]: isn't it a dataset dependency?

**Review by RoriCremer — APPROVED** [review]

**Review by ahaessly — APPROVED** [review]
👍

**Review by thathert — APPROVED** [review]
👍

File: src/main/java/bio/terra/flight/study/delete/DeleteStudyValidateStep.java (hunk 1 above)
Comment (ddietterich) [reply src/main/java/bio/terra/flight/study/delete/DeleteStudyValidateStep.java:38]: Yes, it is!

**Pull request merged**